MyLibrary/BusinessLogic/LibraryService/BookService.cs

```diff
@@ -0,0 +1,337 @@
+            }
+
+            if (!this.dataProvider.Books.Any(b => b.Id == bookId))
+            {
+                throw new ArgumentException("Can`t find book with such id!");
+            }
+
+            if (this.dataProvider.PairsBookAuthor.Any(b => b.BookId == bookId && b.AuthorId == authorId))
+            {
+                throw new ArgumentException("Such book has such author!");
+            }
+
+            BookAuthorPair pairForAdding = new BookAuthorPair { BookId = bookId, AuthorId = authorId };
+            this.dataProvider.AddBookAuthorPair(pairForAdding);
+            this.dataProvider.Save();
+            return pairForAdding.Id;
+        }
+
+        /// <summary>
+        /// Removes the author from book
+        /// </summary>
+        /// <param name="bookId">The book identifier</param>
+        /// <param name="authorId">The author identifier</param>
+        /// <returns>
+        /// The id of removed record in Book-Author pair
+        /// or null if there are any pair of specified book and author
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// Can`t find author with such id!
+        /// or
+        /// Can`t find book with such id!
+        /// </exception>
+        public int? RemoveAuthorFromBook(int bookId, int authorId)
+        {
+            if (!this.dataProvider.Authors.Any(a => a.Id == authorId))
+            {
+                throw new ArgumentException("Can`t find author with such id!");
+            }
+
+            if (!this.dataProvider.Books.Any(b => b.Id == bookId))
+            {
+                throw new ArgumentException("Can`t find book with such id!");
+            }
+
+            BookAuthorPair pairForRemoving = this.dataProvider.PairsBookAuthor.FirstOrDefault(
+                pairBookAuthor => pairBookAuthor.AuthorId == authorId
+                && pairBookAuthor.BookId == bookId);
+            if (pairForRemoving == null)
+            {
+                return null;
+            }
+
+            int removedPairId = pairForRemoving.Id;
+            this.dataProvider.RemoveBookAuthorPair(pairForRemoving);
+            this.dataProvider.Save();
+            return removedPairId;
+        }
+
+        /// <summary>
+        /// Get all books by specified author
+        /// </summary>
+        /// <param name="authorId">The author identifier</param>
+        /// <returns>
+        /// Enumeration of books
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// Can`t find author with such id!
+        /// </exception>
+        public IEnumerable<Book> GetBooksByAuthor(int authorId)
+        {
+            if (!this.dataProvider.Authors.Any(a => a.Id == authorId))
+            {
+                throw new ArgumentException("Can`t find author with such id!");
+            }
+
+            var foundBooks = this.dataProvider.PairsBookAuthor.Where(entry => entry.AuthorId == authorId).
```

Comment:
mm actually here, I suppose it will be okay to skip foundbooks variable and just return.